pkg/infra/baseAgent.go

```diff
@@ -210,7 +210,7 @@ func (a *Base) TakeFood(amountOfFood food.FoodType) (food.FoodType, error) {
 	if amountOfFood < 0 {
 		return 0, &NegFoodError{}
 	}
-	foodTaken := food.FoodType(math.Min(float64(a.tower.currPlatFood), float64(amountOfFood)))
+	foodTaken := food.FoodType(utilFunctions.MinInt(int(a.tower.currPlatFood), int(amountOfFood), int(a.tower.healthInfo.MaxFoodIntake)))
```

Comment:
It means that agents are not allowed to destroy the food anymore (which is what selfish and random agents were doing).

Reply:
Yep, unless we let them destroy food by setting the food intake to 100 or whatever the max is.... think this is a good point for experimentation... Open to opinions tho